courses/management/commands/generate_certificate.py

```diff
@@ -0,0 +1,65 @@
+"""
+Management command to generate certificates
+"""
+from django.core.management.base import BaseCommand, CommandError
+from courses.utils import sync_courserun_grades, sync_courserun_certificates
+
+
+class Command(BaseCommand):
+    """
+    Command to generate certificates for a course.
+    """
+
+    help = "Generates certificates for a course for all enrolled users or a specified user."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--email",
+            type=str,
+            help="The email address of the enrolled User.",
+            required=False,
+        )
+        parser.add_argument(
+            "--course-run",
+            type=str,
+            help="The 'courseware_id' value for a CourseRun.",
+            required=True,
+        )
+        parser.add_argument(
+            "--grade",
+            type=float,
+            help="Override a grade. Setting grade: 0 blacklists a user. Setting a passing grade \
+                (>0.0) whitelists a user. range: 0.0 - 1.0",
```

Comment:
I know this is what I asked for, but I'm concerned whether it covers this typical use case:

1. The course isn't over yet, so certificates haven't been generated for the whole course. A user is blacklisted, but no certificate exists yet, so no action is taken. 

2. When the course end date passes, we run the automated course certificate creation (#877). 

How will we know to skip over the blacklisted user?

Reply:
@pdpinch right now when the certificate is being generated it looks at our local database `CourseRunGrade` record. If, say, we blacklist a user by running `manage.py generate_certificate --email xyz --grade 0.0` it will:
- Generate a `CourseRunGrade` with `grade` value = `0.0`.
- When it goes on to generate a certificate the command will skip over this user because he/she has a grade with value 0.0. Moreover the command will delete any previous `CourseRunCertificate` for this user for the specified course run id that may have existed before.

Reply:
Just to emphasize: any subsequent automated command will also skip this user because they already have a `CourseRunGrade` with the `grade` value equal to `0.0` so their certificate is not generated.

The only way to then reverse that would be to run the management command for that specific `user/courseware_id` combination with the `--force` option set so that it overwrites the existing `CourseRunGrade` record, possibly with a passing value so the certificate will then be generated for this learner.